src/utilities/texture_atlas.jl

```diff
@@ -1,150 +1,253 @@
+    # for things to look clear with a 64px glyph size.
+    glyph_padding::Int32
+    # We save glyphs as signed distance fields, i.e. we save the distance
+    # a pixel is away from the edge of a symbol (continuous at the edge).
+    # To get accurate distances we want to draw the symbol at high
+    # resolution and then downsample to the pix_per_glyph.
+    downsample::Int32
+    font_render_callback::Vector{Function}
 end
 
 Base.size(atlas::TextureAtlas) = size(atlas.data)
+Base.size(atlas::TextureAtlas, dim) = size(atlas)[dim]
 
-@enum GlyphResolution High Low
-
-const HIGH_PIXELSIZE = 64
-const LOW_PIXELSIZE = 32
-
-const CACHE_RESOLUTION_PREFIX = Ref("high")
-const TEXTURE_RESOLUTION = Ref((2048, 2048))
-const PIXELSIZE_IN_ATLAS = Ref(HIGH_PIXELSIZE)
-# Small padding results in artifacts during downsampling. It seems like roughly
-# 1.5px padding is required for a scaled glyph to be displayed without artifacts.
-# E.g. for fontsize = 8px we need 1.5/8 * 64 = 12px+ padding (in each direction)
-# for things to look clear with a 64px glyph size.
-const GLYPH_PADDING = Ref(12)
-
-function set_glyph_resolution!(res::GlyphResolution)
-    if res == High
-        TEXTURE_RESOLUTION[] = (2048, 2048)
-        CACHE_RESOLUTION_PREFIX[] = "high"
-        PIXELSIZE_IN_ATLAS[] = HIGH_PIXELSIZE
-        GLYPH_PADDING[] = 12
-    else
-        TEXTURE_RESOLUTION[] = (1024, 1024)
-        CACHE_RESOLUTION_PREFIX[] = "low"
-        PIXELSIZE_IN_ATLAS[] = LOW_PIXELSIZE
-        GLYPH_PADDING[] = 6
-    end
+"""
+    get_uv_img(atlas::TextureAtlas, glyph_bezierpath, [font])
+
+Helper to debug texture atlas (this usually happens on the GPU)!
+
+can be used like this:
+```julia
+matr = Makie.get_uv_img(atlas, glyph_index, font)
+scatter(Point2f(0), distancefield=matr, uv_offset_width=Vec4f(0, 0, 1, 1), markersize=100)
+```
+"""
+get_uv_img(atlas::TextureAtlas, glyph, font) = get_uv_img(atlas, primitive_uv_offset_width(atlas, glyph, font))
+get_uv_img(atlas::TextureAtlas, path) = get_uv_img(atlas, primitive_uv_offset_width(atlas, path, nothing))
+function get_uv_img(atlas::TextureAtlas, uv_rect::Vec4f)
+    xmin, ymin, xmax, ymax = round.(Int, uv_rect .* Vec4f(size(atlas)..., size(atlas)...))
+    xmax = xmax + 1
+    ymax = ymax + 1
+    return atlas.data[Rect(xmin, ymin, xmax - xmin, ymax - ymin)]
 end
 
-function TextureAtlas(initial_size = TEXTURE_RESOLUTION[])
+function TextureAtlas(; resolution=2048, pix_per_glyph=64, glyph_padding=12, downsample=5)
     return TextureAtlas(
-        RectanglePacker(Rect2(0, 0, initial_size...)),
-        Dict{Tuple{UInt64, String}, Int}(),
-        1,
+        RectanglePacker(Rect2{Int32}(0, 0, resolution, resolution)),
+        Dict{UInt32, Int}(),
         # We use float max here to avoid texture bleed. See #2096
-        fill(Float16(0.5PIXELSIZE_IN_ATLAS[] + GLYPH_PADDING[]), initial_size...),
+        fill(Float16(0.5pix_per_glyph + glyph_padding), resolution, resolution),
```

Comment:
Seems like I didn't update this when I switched away from floatmax. 

The fill value there is the maximum distance from the center of a glyph. With stroke- and glowwidth the background can become relevant during rendering and floatmax to some value can become a harsh cut-off. With this intermediate value it's less harsh (though still there). Basically this is closer to the values you'd get with more padding, which gives you softer transitions with large glow/strokewidths.